project-base/storefront/components/Blocks/Product/DeferredPromotedProducts.tsx

```diff
@@ -16,24 +18,32 @@ const ProductsSliderPlaceholder = dynamic(() =>
 );
 
 export const DeferredPromotedProducts: FC = () => {
+    const { t } = useTranslation();
     const [{ data: promotedProductsData, fetching: arePromotedProductsFetching }] = usePromotedProductsQuery();
     const shouldRender = useDeferredRender('promoted_products');
 
-    if (arePromotedProductsFetching) {
-        return <SkeletonModulePromotedProducts />;
-    }
-
-    if (!promotedProductsData?.promotedProducts) {
+    if (!promotedProductsData?.promotedProducts.length) {
```

Comment:
Same question, does the skeleton loader works with this change?

Reply:
fixed